apps/explorer/lib/explorer/chain.ex

```diff
@@ -1726,6 +1726,55 @@ defmodule Explorer.Chain do
       0
   end
 
+  @doc """
+  Fetches count of last n blocks, last block timestamp, last block number and average gas used in the last minute.
+  Using a single method to fetch and calculate these values for performance reasons (only 2 queries used).
+  """
+  @spec metrics_fetcher(integer | nil) ::
+          {non_neg_integer, non_neg_integer, non_neg_integer, float}
+  def metrics_fetcher(n) do
+    {last_block_number} =
+      Repo.one(
```

Comment:
The query for finding of max consensus block number in the DB has been already defined in `Chain.fetch_max_block_number/0` method. I suppose you can re-use this method here.